packages/blockchain-link-utils/src/solana.ts

```diff
@@ -66,16 +74,49 @@ export const getTokenNameAndSymbol = (mint: string, tokenDetailByMint: TokenDeta
           };
 };
 
+type SplTokenAccountData = {
+    /** Name of the program that owns this account */
+    program: 'spl-token';
+    /** Parsed account data */
+    parsed: {
+        info: {
+            mint: string;
+            tokenAmount: {
+                amount: string;
+                decimals: number;
+            };
+        };
+    };
+    /** Space used by account data */
+    space: number;
+};
+
+type SplTokenAccount = { account: AccountInfo<SplTokenAccountData>; pubkey: PublicKey };
+
+const isSplTokenAccount = (tokenAccount: ApiTokenAccount): tokenAccount is SplTokenAccount => {
+    const { parsed } = tokenAccount.account.data;
+    return (
+        tokenAccount.account.data.program === 'spl-token' &&
+        'info' in parsed &&
+        'mint' in parsed.info &&
+        typeof parsed.info.mint === 'string' &&
+        'tokenAmount' in parsed.info &&
+        typeof parsed.info.tokenAmount.amount === 'string' &&
+        typeof parsed.info.tokenAmount.decimals === 'number'
+    );
+};
+
 export const transformTokenInfo = (
     tokenAccounts: ApiTokenAccount[],
     tokenDetailByMint: TokenDetailByMint,
 ) => {
+    // since ApiTokenAccount type is not precise enough, we type-guard the account to make sure they contain all the necessary data
+    const validTokenAccounts = tokenAccounts.filter(isSplTokenAccount);
```

Comment:
Not very important, but given that the original author had already used FP approach, it somehow feels like the `isSplTokenAccount` filtering should be part of the pipe below.

Reply:
Done, I rebase and force-pushed so you can find the change in [fix(blockchain-link): filter out non-spl tokens](https://github.com/trezor/trezor-suite/pull/10437/commits/39212cf9a58b8883fcf0c385527d052e1bdffc5c)